api/core/span_context.go

```diff
@@ -49,72 +68,82 @@ func TraceIDFromHex(h string) (TraceID, error) {
 		return t, errors.New("hex encoded trace-id must have length equals to 32")
 	}
 
+	if err := decodeHex(h, t[:]); err != nil {
+		return t, err
+	}
+
+	if !t.IsValid() {
+		return t, errors.New("trace-id can't be all zero")
+	}
+	return t, nil
+}
+
+// TraceIDFromHex returns a SpanID from a hex string if it is compliant
+// with the w3c trace-context specification.
+// See more at https://www.w3.org/TR/trace-context/#parent-id
+func SpanIDFromHex(h string) (SpanID, error) {
+	s := SpanID{}
+	if len(h) != 16 {
+		return s, errors.New("hex encoded span-id must have length equals to 16")
```

Comment:
make this and other errors  a static error.

Reply:
changed